test/terra/backends/test_pulse_simulator.py

```diff
@@ -0,0 +1,806 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+PulseSimulator Integration Tests
+"""
+
+import unittest
+from test.terra import common
+
+import numpy as np
+from scipy.linalg import expm
+
+import qiskit
+import qiskit.pulse as pulse
+
+from qiskit.compiler import assemble
+
+from qiskit.test.mock import FakeOpenPulse2Q
+from qiskit.pulse.commands import SamplePulse, FrameChange, PersistentValue
+
+class TestPulseSimulator(common.QiskitAerTestCase):
+    r"""PulseSimulator tests.
+
+    Mathematical expressions are formulated in latex in docstrings for this class.
+
+    # pylint: disable=anomalous backslash in string
+    Uses single qubit Hamiltonian `H = -\frac{1}{2} \omega_0 \sigma_z + \frac{1}{2} \omega_a
+    e^{i(\omega_{d0} t+\phi)} \sigma_x`. We make sure H is Hermitian by taking the complex conjugate
+    of the lower triangular piece (as done by the simulator). To find the closed form, we apply move
+    to a rotating frame via the unitary `Urot = e^{-i \omega t \sigma_z/2}`. In this frame, the
+    Hamiltonian becomes `Hrot = \frac{1}{2} \omega_a (\cos(\phi) \sigma_x - \sin(\phi) \sigma_y)
+    + \frac{\omega_{d0}-\omega_0}{2} \sigma_z`.
+    """
+
+    def setUp(self):
+        """ Set configuration settings for pulse simulator """
+        qiskit.IBMQ.load_account()
```

Comment:
Do we still need to connect to a real backend to run the pulse sim?
Otherwise let's not load credentials.

Reply:
Another option could be mocking this (get a real response and hardconding it)

Reply:
Yup fixed it!